srcCpp/RTIDDSLoggerDevice.h

```diff
@@ -0,0 +1,71 @@
+#ifndef __RTIDDSLOGGERDEVICE_H__
+#define __RTIDDSLOGGERDEVICE_H__
+
+/*
+ * (c) 2005-2018  Copyright, Real-Time Innovations, Inc. All rights reserved.
+ * Subject to Eclipse Public License v1.0; see LICENSE.md for details.
+ */
+
+#include <string>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ndds/ndds_cpp.h"
+
+/*
+ *  After setting "UPDv4 | SHMEM" as default transport_builtin mask. The user may
+ *  get the following error running a default scenario:
+ *
+ *      [D0001|ENABLE]NDDS_Transport_Shmem_create_recvresource_rrEA:failed to initialize shared memory resource segment for key 0x40894a
+ *      [D0001|ENABLE]NDDS_Transport_Shmem_create_recvresource_rrEA:failed to initialize shared memory resource segment for key 0x40894c
+ *      [D0001|ENABLE]DDS_DomainParticipantPresentation_reserve_participant_index_entryports:!enable reserve participant index
+ *      [D0001|ENABLE]DDS_DomainParticipant_reserve_participant_index_entryports:Unusable shared memory transport. For a more in-depth explanation of the possible problem and solution, please visit http://community.rti.com/kb/osx510.
+ *      [D0001|ENABLE]DDS_DomainParticipant_enableI:Automatic participant index failed to initialize. PLEASE VERIFY CONSISTENT TRANSPORT / DISCOVERY CONFIGURATION.
+ *      [NOTE: If the participant is running on a machine where the network interfaces can change, you should manually set wire protocol's participant id]
+ *      DDSDomainParticipant_impl::createI:ERROR: Failed to auto-enable entity.
+ *
+ *  By using our own implementation of LoggerDevice, we can filter those errors.
+ *  In the case that those errors appear, the execution will be stopped and
+ *  a message, with two approaches, will be printed:
+ *      - A link (http://community.rti.com/kb/osx510) about how to solve the issue
+ *      - Explain how to set another different transport through command line parameter.
```

Comment:
- How to setup a different transport via command-line parameters

Reply:
Done